### DISCUSSION TITLE
Kong with AWS ALB ingress controller

### DISCUSSION THREAD

**Discussion by alexanderccc** [Kong/kong#9873] [Help]
Hi All,

I'm running a bunch of Kong ingress controllers on Kubernetes and expose them by setting an ingress on the proxy service that's managed by the aws alb controller. The setup has been stable and works well, but to get the health check to pass I need to tell the ALB to consider a status code of 404 as successful as the proxy doesn't have a health endpoint. I could expose the status port as an additional nodePort and point the health check to that, but that would complicate the setup (for a bunch of controllers). 

The Ingress pointing to the proxy service is important here as that's the ALB spec for handling application traffic as I know this could be avoided by exposing the service using an NLB with the LoadBalancer service type setup and route TCP traffic but I'd like to avoid that. 

Any ideas on this? Why does Kong not have a /kong/health or similar endpoint on all it's services (status/admin/proxy)?

**Comment by fffonion** [comment]
@alexanderccc There's a `/health` endpoint for admin endpoint, it should be the same one as status port.
For proxy, it might not be a good idea to reserve `/health` because it could be overwrite upstream services.
Consider using a request-termination plugin that returns 200 to serve as a health check endpoint.
